airflow/contrib/operators/kubernetes_operator.py

```diff
@@ -0,0 +1,126 @@
+        the operator waits for the pod to start running to ensure
+        successful creation.
+    :type wait: boolean
+    :param cleanup: Perform cleanup on completion.
+        Allowed values: Always, Never, OnSuccess, OnFailure.
+        Default Always. Settign wait == False forces Never,
+        as cleanup can be only performed on terminated container.
+    :param labels: Labels and presets to apply to the pod.
+    :type labels: dict
+    :param conn_id: Id of pre-defined k8s connection. Currently not used,
+        as only preconfigured environment with kube config or env variables
+        is supported.
+    :type conn_id: string
+    :param poke_interval: Interval between checking the status in seconds
+    :type poke_interval: integer
+    """
+    template_fields = ('name', 'command', 'op_args', "namespace")
+    ui_color = '#f0ede4'
+
+    @apply_defaults
+    def __init__(
+            self,
+            name,
+            namespace,
+            image,
+            command=None,
+            op_args=None,
+            wait=True,
+            cleanup="Always",
+            labels=None,
+            conn_id="k8s_default",
+            poke_interval=3,
+            *args, **kwargs):
+        super(KubernetesPodOperator, self).__init__(*args, **kwargs)
+        self.image = image
+        self.name = name
+        self.namespace = namespace
+        self.command = command
+        self.op_args = op_args
+        self.wait = wait
+        self.cleanup = cleanup if self.wait else "Never"
+        self.labels = labels
+        self.poke_interval = poke_interval
+        self.conn_id = conn_id
+
+    def _create_hook(self):
+        return KubernetesHook(self.conn_id)
+
+    def should_do_cleanup(self, status):
+        return ((self.cleanup == "Always") or
+                ((self.cleanup == "OnFailure") and (status == "Failed")) or
+                ((self.cleanup == "OnSuccess") and (status == "Succeeded")))
+
+    def execute(self, context):
+        exit_statuses = ["Succeeded"] if self.wait else ["Running", "Succeeded"]
+
+        hook = self._create_hook()
+
+        pod = hook.get_pod_definition(
+            image=self.image,
+            name=self.name,
+            namespace=self.namespace,
+            restart_policy="Never",
+            command=self.command,
+            args=self.op_args,
+            labels=self.labels)
+
+        logging.info("Creating pod %s in namespace %s with following definition: %s",
+                self.name, self.namespace, pod)
+
+        hook.create_pod(pod)
+
+        try:
+            status = None
+            while not (status in exit_statuses):
+                status = hook.get_pod_state(pod)
+                logging.info("Checking pod status => %s", status)
+
+                if (status == "Failed"):
+                    raise AirflowException("Pod failed!")
+
+                sleep(self.poke_interval)
+        finally:
+            if (self.wait and self.should_do_cleanup(status)):
```

Comment:
The loop should not exit in a normal way if the status is None. The only way it can happen is an uncaught exception - in this case the whole DAG blows up anyways, I was thinking what to do in such case - simplest way is to set status to Failed, but thats a lazy solution. A good suggestion welcome!